### ISSUE TITLE
[IAFeedback] Show App Store Review Dialog

### ISSUE THREAD

**Issue by shiki** [woocommerce/woocommerce-ios#2543]
The #2357 task should be completed first. 

<img src="https://user-images.githubusercontent.com/198826/88209433-f1b57c00-cc0f-11ea-885f-443f0a7a4e4a.png" width="240">

Pressing on the **I Like It** button should show the App Store Review dialog. 

We will use `SKStoreReviewController`. No UI customizations are needed since the dialog cannot be customized. But please double-check if this is still true.

## Others

- [ ]  Must be hidden behind feature flag

**Comment by Ecarrion** [comment]
It's worth noticing that while our rules to display the in-app feedback banner seem to match the ones expressed on the internet about `SKStoreReviewController`. It's possible that after tapping the "I like button" the app store review prompt doesn't appear to the user. That's why Apple doesn't recommend linking that `requestReview()` action to a particular user input. 

Are we ok with that edge case/possible scenario? cc @shiki @Garance91540 @astralbodies 

ref: https://developer.apple.com/documentation/storekit/skstorereviewcontroller/requesting_app_store_reviews

----

Alternatively, we can redirect the user to the woo's app store page with a "write-review" parameter that is guaranteed to work 100% of the time, but is not an in-app experience.

![app-store](https://user-images.githubusercontent.com/562080/89078649-685a2400-d34a-11ea-8cbe-b61dc3151ac2.gif)

**Comment by Garance91540** [comment]
Hi @Ecarrion , 

>It's possible that after tapping the "I like button" the app store review prompt doesn't appear to the user. 

In that case, is it possible to simply open the WooCommerce Apple page? https://apps.apple.com/us/app/woocommerce/id1389130815

**Comment by Ecarrion** [comment]
Hey @Garance91540,

We can't tell if the in-app review form was displayed(or not) in order to redirect the user to the app store page 😢 
Also, there is a chance that the user has this setting disabled, so no in-app review form will appear ever.

![IMG_E536B31F940E-1](https://user-images.githubusercontent.com/562080/89222450-b06f8580-d59a-11ea-949f-6464bcba63da.jpeg)

**Comment by Garance91540** [comment]
ah that's really annoying! I didn't realize that was even possible...

Could we do:
1. user select "I like it"
2. we open modal "would you mind leaving us a review? yes/no"
3. user selects yes > open https://apps.apple.com/us/app/woocommerce/id1389130815

let me know if this is possible and I'll update the designs accordingly...

**Comment by Ecarrion** [comment]
@Garance91540 
> ah that's really annoying! I didn't realize that was even possible...
> 
> Could we do:
> 
> 1. user select "I like it"
> 2. we open modal "would you mind leaving us a review? yes/no"
> 3. user selects yes > open https://apps.apple.com/us/app/woocommerce/id1389130815
> 
> let me know if this is possible and I'll update the designs accordingly...

Yes, that seems feasible!

**Comment by Ecarrion** [comment]
@Garance91540 Thinking about it maybe we could just hide the in-app feedback banner once the user taps on "I like it", record some analytics and try our best to show the in-app review alert.

Im thinking that in the case that the in-app review alert don't appear the user will still think that something was done as the in-app banner would disappear.

What do you think?

**Comment by shiki** [comment]
Hey @Ecarrion and @Garance91540. I'm sorry I didn't get to document this more. In CGPNUU63E-p1595382710021100-slack, we discussed this a bit. 

- We are simply limited to calling `function` to show the dialog. And it is up to iOS whether the function call with present the dialog or not. It is also rate-limited (a few dialogs per year or something).
- There is no API to tell whether iOS will or can show the dialog. 

We are simply limited to calling a `function`. 😞 

For these reasons, we decided that pressing *I Like It* will **dismiss the feedback card right away**.

--- 

But let us know, @Garance91540, if you feel that this was a wrong decision. 🙂
